Sgdemux

sam-white04 opened the pull request:
Opening a draft PR for increased readability of changes to `nf-core-modules/modules/nf-core/sgdemux/` & `nf-core-modules/tests/modules/nf-core/sgdemux`

Commit:
First commit sgdemux

Commit:
First commit to sgdemux

Commit:
Update modules/nf-core/sgdemux/meta.yml

Commit:
Added TODO modules/nf-core/sgdemux/main.nf

nh13 commented on modules/nf-core/sgdemux/main.nf:
```diff
@@ -0,0 +1,49 @@
+    if (workflow.profile.tokenize(',').intersect(['Singularity', 'Docker']).size() >= 1) {
+        exit 1, "SGDEMUX module does not support Singularity or Docker. Please use Conda / Mamba instead."
+    }
+    // TODO check sgdemux is not on singularity/docker (google says no) double check with Nils
```
Having `sgdemux` in `bioconda` will make a docker container accessible on `quay.io`.  See https://bioconda.github.io/recipes/sgdemux/README.html#package-sgdemux, in particular:
`quay.io/biocontainers/sgdemux:<tag>`

For singularity, we'll need a `Dockerfile` ([see requirement (5)](https://nf-co.re/developers/modules#software-requirements)).  Feel free to ask the team how to create a dockerfile (or look for examples in other modules in nf-core/modules)

nh13 replied:
Feel free to ping the nf-core folks on slack for help on this too!

sam-white04 replied:
I see `sgdemux` on `quay.io`, and I will make changes to line 12 in the next commit to reflect that. I am happy to create a `Dockerfile` for `sgdemux`. I did not know that was up to us

sam-white04 replied:
@nils I am a bit confused about these container requirements. Requirement 5 in the link you provided makes me think that a `Dockerfile` must be provided only if software ifsnot available on `bioconda`. So if sgdemux is already available via `bioconda` and  `quay.io`, do we need `singularity`?

nh13 commented on modules/nf-core/sgdemux/main.nf:
```diff
@@ -0,0 +1,49 @@
+        // 'quay.io/biocontainers/YOUR-TOOL-HERE' }"]
+
+    input:
+    tuple val(meta), path(run_manifest), path(run_dir)
```
```suggestion
    tuple val(meta), path(sample_sheet), path(in_fastqs_dir)
```

nh13 replied:
I'd also add a comment either here or above the process that the input FASTQs must be bgzipped!

sam-white04 replied:
Agreed, in fact I already added that locally before seeing this.  Should we add some type of check to ensure fastq files are in the correct format?

nh13 commented on modules/nf-core/sgdemux/main.nf:
```diff
@@ -0,0 +1,49 @@
+    tuple val(meta), path(run_manifest), path(run_dir)
+
+    output:
+    // TODO output/Samples/*/*_R*...
```
```suggestion
```

nh13 commented on modules/nf-core/sgdemux/main.nf:
```diff
@@ -0,0 +1,49 @@
+    tuple val(meta), path('output/per_project_metrics.tsv'), emit: per_project_metrics
+    tuple val(meta), path('output/per_sample_metrics.tsv'), emit: per_sample_metrics
+    tuple val(meta), path('output/sample_barcode_hop_metrics.tsv'), emit: sample_barcode_hop_metrics
+    path "versions.yml"           , emit: versions
```
align emits

nh13 commented on modules/nf-core/sgdemux/main.nf:
```diff
@@ -0,0 +1,49 @@
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def runManifest = run_manifest ? "-r ${run_manifest}" : ""
```
```suggestion
    def sampleSheet = run_manifest ? "-r ${run_manifest}" : ""
```

nh13 commented on modules/nf-core/sgdemux/main.nf:
```diff
@@ -0,0 +1,49 @@
+        --output-dir output/ \\
+        --demux-threads $task.cpus \\
+        --compressor-threads $task.cpus \\
+        --writer-threads $task.cpus \\
```
```suggestion
        --writer-threads $task.cpus
```

nh13 commented on modules/nf-core/sgdemux/meta.yml:
```diff
@@ -0,0 +1,64 @@
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - run_manifest:
```
Can you use sample sheet here and in the descriptionb elow

nh13 commented on modules/nf-core/sgdemux/meta.yml:
```diff
@@ -0,0 +1,64 @@
+      type: file
+      description: RunManifest file
+      pattern: "*.{csv}"
+  - run_dir:
```
```suggestion
  - fastqs_dir:
```

nh13 commented on modules/nf-core/sgdemux/meta.yml:
```diff
@@ -0,0 +1,64 @@
+      pattern: "*.{csv}"
+  - run_dir:
+      type: directory
+      description: "Input run directory containing optionally containing a RunManifest.json if run_manifest is not supplied"
```
Please update based on docs.  This should be a directory containing the FASTQs.

nh13 commented on modules/nf-core/sgdemux/meta.yml:
```diff
@@ -0,0 +1,64 @@
+      pattern: "versions.yml"
+  - sample_fastq:
+      type: file
+      description: Demultiplexed sample FASTQ files
```
```suggestion
      description: Demultiplexed per-sample FASTQ files
```

nh13 commented on modules/nf-core/sgdemux/meta.yml:
```diff
@@ -0,0 +1,64 @@
+      type: file
+      description: Demultiplexed sample FASTQ files
+      pattern: "output/*_R*.fastq.gz"
+  - metrics:
```
update here and below based on https://github.com/Singular-Genomics/singular-demux#metrics

nh13 commented on modules/nf-core/sgdemux/nextflow.config:
```diff
@@ -0,0 +1,46 @@
+params {
```
I don't think we need this file.  I don't see it in `fastqc` or `bcl2fastq` for example.  Thoughts?

sam-white04 replied:
You are correct, `fastqc` nor `bcl2fastq` have `nextflow.config` files I will remove this file for sgdemux.

nh13 commented on tests/modules/nf-core/sgdemux/main.nf:
```diff
@@ -0,0 +1,16 @@
+    // input = Channel.value([
+    //    [ id: 'sim-data' ], // meta map
+    //    file("/Users/.../projects/SINGULAR/singular-demux/simulated/out.sample_meta.csv", checkIfExists: true),
+    //    file("/Users/.../projects/SINGULAR/singular-demux/simulated/", checkIfExists: true)
```
TODO: add a tarball to `nf-core/test-datasets`, then have a process to untar it prior to running SGDEMUX, similar to: https://github.com/nf-core/modules/blob/master/tests/modules/nf-core/bases2fastq/main.nf

nh13 commented on tests/modules/nf-core/sgdemux/nextflow.config:
```diff
@@ -0,0 +1,5 @@
+process {
+
+    publishDir = { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" }
+    
```
as per bases2fastq 
```suggestion
     withName: UNTAR {
        publishDir = [ enabled: false ]
    }
```

nh13 commented on tests/modules/nf-core/sgdemux/test.yml:
```diff
@@ -0,0 +1,11 @@
+## TODO nf-core: Please run the following command to build this file:
+#                nf-core modules create-test-yml sgdemux
```
```suggestion
```

jfy133 committed:
Update: DAS_Tool to 1.1.6 (#2765)

* Bump das_tool versions

* Roll back scaffolds2binversion as this is a legacy

* Correct build container

* Missing quote

matthdsm committed:
New module: Somalier ancestry (#2760)

* first commit somalier_ancestry

* add output files

* add output files

* path changed to directory

* first test

* add some config changes

* update module structure

* typo's

* fix test

* Update test.yml

* Update meta.yml

* indented emits

* improved readability

* copy paste error

* make sure the module runs

* syntax fix

* query input files in nf-tests directly

* update test

* md5sum was variable, modified contains section

* style changes

* Update tests/modules/nf-core/somalier/ancestry/main.nf

Co-authored-by: Jonathan Manning <jonathan.manning@healx.io>

* Revert "style changes"

This reverts commit 25d6d02c5494d6a7b582971487204b53396ffb14.

Co-authored-by: Anabella Trigila <anatrigila@gmail.com>
Co-authored-by: Jonathan Manning <jonathan.manning@healx.io>

Joon-Klaps committed:
[FEATURE] update trimmomatic to emit summary output  (#2748)

* adding summary output

* added description of summary file

* updating test.yml files

* Updating the test.yml file to fit previous layout

* fastq md5sum are variable across containers

* adding stderr for mqc

* comment last test

* fix typo

* updating test.ym files including stderr

* [automated] Fix linting with Prettier

* Revert commits back to e050ad7

* style: Fix indention

Co-authored-by: Joon-Klaps <joon.klaps@gmail.com>
Co-authored-by: nf-core-bot <core@nf-co.re>
Co-authored-by: Edmund Miller <edmund.a.miller@protonmail.com>

Commit:
Updated modules/../main.nf and meta.yml

Commit:
Update software requirements main.nf

Commit:
Fix Typo

Commit:
Illustrate somalier/ancestry tidy-up (#2773)

Commit:
Update tests

Commit:
Update main.nf

Commit:
Update test.yml

Commit:
Update main.nf

Commit:
Update main.nf

Commit:
untar: need a gnu grep to use -P, also use a consistent tar (#2772)

* need a gnu grep to use -P in untar

* Update main.nf

Commit:
Update file paths to forked test-datasets

Commit:
Updated test.yml

Commit:
Update test.yml with md5sum for all outputs

Commit:
Update main.nf with url to nf-core/test-datasets

LouisLeNezet committed:
Glimpse phasing modules (#2766)

* New Glimpse phasing modules

* Revert modification of pytest_modules.yml

* Take out trailing spaces

* Missing final new line

* Delete params.enable_conda

* Delete parenthesis

* Delete unecessary new lines

* Update modules/nf-core/glimpse/phase/main.nf

Co-authored-by: Nicolas Vannieuwkerke <101190534+nvnieuwk@users.noreply.github.com>

* Simplify phase module by only region

* Add index input files and delete region_file parameter

* Update test and add new one for sample information

* Correct spelling mistake

* Fix linting

* Create sample file on the fly

* Input and Output regions in input tuple for parallelization

* Simplify version control

* Rename variable for comprehension

Co-authored-by: Nicolas Vannieuwkerke <101190534+nvnieuwk@users.noreply.github.com>

LouisLeNezet committed:
Update glimpse chunk (#2774)

Update glimpse chunk for better parallelization and version control

jfy133 committed:
Update: fix bug in conda scope check for antismash download databases (#2776)

* Bump das_tool versions

* Roll back scaffolds2binversion as this is a legacy

* Correct build container

* Missing quote

* Fix bug in conda scope check condition for antismash download databases

* Update main.nf

heuermh committed:
Adding new module smoothxg (#1150)

* Adding new module smoothxg.

* fix conda version

* move to nf-core directory

* apply prettier

* fix test paths

* update to version 0.6.5, fix linting

* fix tests, add version

* also test with actual pangenome data

* add optional MAF output

* add test with recurse

* missed test declaration

* enable recursion preview feature

* remove recursion test

* update smoothxg to version 0.6.7

* refactor to smoothxg_once and smoothxg_iterate

* Revert "refactor to smoothxg_once and smoothxg_iterate"

This reverts commit fc32ea9b2f3178df99c4b20ce2cad740659d6985.

* update container image

* Update smoothxg to 0.6.8.

* specify --n-haps to fix the tests

* enable conda by default

Co-authored-by: James A. Fellows Yates <jfy133@gmail.com>
Co-authored-by: Simon Heumos <simon.heumos@qbic.uni-tuebingen.de>

Commit:
All lints are passing except container versions

Commit:
Re-set container/conda commands from nf-core modules create...

Commit:
Galaxy container place-holder

Commit:
Add sgdemux galaxy place holder

Commit:
bump seqwish to v.0.7.8 (#2780)

Commit:
Add new module `odgi squeeze` (#2781)

* engange odgi squeeze!

* EOL

* remove dangling TODO

* Update modules/nf-core/odgi/squeeze/main.nf

Co-authored-by: Jonathan Manning <jonathan.manning@healx.io>

* Update tests/modules/nf-core/odgi/squeeze/main.nf

Co-authored-by: Jonathan Manning <jonathan.manning@healx.io>

* align emits

* fix tests, don't use Channel.from()

Co-authored-by: Jonathan Manning <jonathan.manning@healx.io>

Commit:
Update odgi modules to v0.8.2 (#2785)

bump odgi modules to v0.8.2

jasmezz committed:
Fix bakta/baktadbdownload (#2786)

* Fix database output path

jasmezz committed:
Fix bakta/bakta (#2787)

* Fix bakta command in script block

maxulysse committed:
Fix sarek issues (#2784)

Commit:
Bump `wfmash` to v0.10.2 (#2778)

* add module wfmash

* more I/O query options

* Update modules/nf-core/wfmash/main.nf

Co-authored-by: Adam Talbot <12817534+adamrtalbot@users.noreply.github.com>

* Update modules/nf-core/wfmash/main.nf

Co-authored-by: Adam Talbot <12817534+adamrtalbot@users.noreply.github.com>

* address review comments

* bump wfmash to 0.10.2

Co-authored-by: Adam Talbot <12817534+adamrtalbot@users.noreply.github.com>
Co-authored-by: Maxime U. Garcia <max.u.garcia@gmail.com>

jasmezz committed:
Bakta version bump to 1.6.1 (#2788)

* Add knownclusterblast directory as output channel

* Version bump to 1.6.1

WackerO committed:
Shinyngs/validator: Make feature file optional (#2531)

* Separated inputs and outputs and marked feature_meta as optional; local
module test fails but so it does on the master branch

* Hopefully fixed the error

* Put assays back into a tuple with sample; fixed no_feature test and added yml entry

* Update modules/nf-core/shinyngs/validatefomcomponents/meta.yml

Co-authored-by: Jonathan Manning <jonathan.manning@healx.io>

* Update modules/nf-core/shinyngs/validatefomcomponents/meta.yml

Co-authored-by: Jonathan Manning <jonathan.manning@healx.io>

* Update modules/nf-core/shinyngs/validatefomcomponents/meta.yml

Co-authored-by: Jonathan Manning <jonathan.manning@healx.io>

Co-authored-by: Jonathan Manning <jonathan.manning@healx.io>

LouisLeNezet committed:
New module: glimpse ligate (#2779)

* Make the test for glimpse ligate

* Add glimpse ligate to the pytest_modules.yml

* Set vcf format as default

* Add change of format to meta.yml

* Fix typo

* Add compressed files output

* Add compressed files output

* Set vcf.gz as default output

LouisLeNezet committed:
Update Glimpse phase for compressed output (#2794)

* Add launch

* Add compressed file format for output

* Delete launch.sh

* Change file names in test.yml

LouisLeNezet committed:
New module Glimpse sample (#2795)

* Add launch

* Correct variable names in test.yml and main of ligate

* Delete unecessary new line

* Add all files for new modules Glimpse Sample

* Add possibility of compressed files as input in glimpse chunk

* Delete launch file

* [automated] Fix linting with Prettier

---------

Co-authored-by: nf-core-bot <core@nf-co.re>

BioInf2305 committed:
add a new module sickle (#2757)

* add a new module sickle

* removed qual from meta and addded it as a separate input, updated meta.yml

* ran prettier formatting on meta.yml

---------

Co-authored-by: Nicolas Vannieuwkerke <101190534+nvnieuwk@users.noreply.github.com>

zachary-foster committed:
New module: Graphtyper/genotype (#2727)

Adds module for `genotype` command from `graphtyper` and associated tests. 

---------

Co-authored-by: Nicolas Vannieuwkerke <101190534+nvnieuwk@users.noreply.github.com>

Commit:
Make untar work for non-gzipped tars (#2796)

* Use -a flag for flexible decompression

* improve stub

Co-authored-by: Adam Talbot <12817534+adamrtalbot@users.noreply.github.com>

* Use prefix

* simpleName -> baseName

* Rever to simpleName

* Better prefix use

* Fix stub as well

* align equals

* Override prefix to fix kraken error

* revert unnecessary change

* Use archive baseName when neither prefix nor meta id available

* Revert bad edit

* Simplify- regex will do it

* Don't forget to match end of string in regex

* Match stub prefix

* Don't need kraken2 config change now

* Alternate extension handling

Co-authored-by: Matthieu Muffato <cortexspam-github@yahoo.fr>

---------

Co-authored-by: Adam Talbot <12817534+adamrtalbot@users.noreply.github.com>
Co-authored-by: Matthieu Muffato <cortexspam-github@yahoo.fr>

Commit:
Fixed container error

Commit:
Bump shinyngs to address palette issues (#2798)

* Bump shinyngs to address https://github.com/nf-core/differentialabundance/issues/56

* tidy up singularity links

Commit:
Fixed md5sum errors

sam-white04 commented:
Here is the latest version of [sgdemux](https://github.com/fulcrumgenomics/nf-core-modules/commit/be297803c88936c8a665fa52e80199a1234371a8), including the checks shown below. Most importantly `test.yml` has been updated to exclude md5sums for fastq's fue to sorting errors. 

## PR checklist

- [x] This comment contains a description of changes (with reason).

- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

Commit:
Add affy/justrma module  (#2790)

* Add module starting file complement

* prettier

* no SHA for .rds

* Refer to archive instead of individual cels

* Decide untar flags from exension

* Add templates dir to git

* Remove rogue comma

* appease eclint

* Revert "Decide untar flags from exension"

This reverts commit 82c027498ef230521a085363a6448c4020c3df5d.

* Use -a flag for flexible decompression

* improve stub

Co-authored-by: Adam Talbot <12817534+adamrtalbot@users.noreply.github.com>

* Use prefix

* simpleName -> baseName

* Rever to simpleName

* Better prefix use

* Fix stub as well

* Env fixes

* CDF install fixes

* Minor fix

* Test bioconda preprocesscore fix

* template fixes

* apply config

* Update test files

* round results, update tests

* Don't check md5sum on eset.rds

* Revert config change

* simplify in light of upstream fixes

* Meta fixes

* Remove artifact

* Re-point the test data properly

* Document CDF install function

* Appease eclint

---------

Co-authored-by: Adam Talbot <12817534+adamrtalbot@users.noreply.github.com>

nh13 commented on modules/nf-core/sgdemux/meta.yml:
```diff
@@ -0,0 +1,68 @@
+      description: "Tool for demultiplexing sequencing data generated on
+      Singular Genomics' sequencing instruments."
+      homepage: "https://github.com/Singular-Genomics/singular-demux"
+      documentation: "https://github.com/Singular-Genomics/singular-demux#per_project_metricstsv"
```
Top of readme?
```suggestion
      documentation: "https://github.com/Singular-Genomics/singular-demux#sgdemux"
```

nh13 commented on modules/nf-core/sgdemux/meta.yml:
```diff
@@ -0,0 +1,68 @@
+        e.g. [ id:'test', single_end:false ]
+  - sample_sheet:
+      type: file
+      description: sample_sheet file (two column csv wih Sample_Barcode and Sample_ID)
```
```suggestion
      description: sample_sheet file (either a Singular Genomics sample sheet, or a two column csv wih Sample_Barcode and Sample_ID)
```

nh13 commented on modules/nf-core/sgdemux/meta.yml:
```diff
@@ -0,0 +1,68 @@
+      pattern: "*.{csv}"
+  - fastqs_dir:
+      type: directory
+      description: "Input directory containing bgzipped FASTQ files"
```
```suggestion
      description: "Input directory containing bgzipped (not gzip) FASTQ files"
```

nh13 reviewed (approved):
Minor comments, looks good.  Fixup and then retarget to the main nf-core/modules repo!

sam-white04 committed:
Apply suggestions from code review

Co-authored-by: Nils Homer <nh13@users.noreply.github.com>

nh13 committed:
Update modules/nf-core/sgdemux/meta.yml

nh13 committed:
Apply suggestions from code review

nh13 committed:
Update modules/nf-core/sgdemux/main.nf

nh13 committed:
prettier

nh13 committed:
prettier again

adamrtalbot committed:
new module: cnvkit/call (#2811)

- Adds cnvkit/call module which can be used to call CNR from cnvkit/batch.
- Includes ability to use a VCF file for allele fractions.
- samtools added as dependency which appears to fix some conda stuff.

nvnieuwk committed:
new module vg/construct (#2813)

* new module vg/construct

* Update modules/nf-core/vg/construct/meta.yml

Co-authored-by: Jonathan Manning <jonathan.manning@healx.io>

* add insertions to msa

* fix if statement

* update to the input handling

* Update modules/nf-core/vg/construct/main.nf

* Update modules/nf-core/vg/construct/meta.yml

* Update modules/nf-core/vg/construct/meta.yml

Co-authored-by: Adam Talbot <12817534+adamrtalbot@users.noreply.github.com>

---------

Co-authored-by: Jonathan Manning <jonathan.manning@healx.io>
Co-authored-by: Adam Talbot <12817534+adamrtalbot@users.noreply.github.com>

sam-white04 committed:
Removed should_exist: true

nh13 committed:
prettier

sam-white04 committed:
Merge branch 'master' into sgdemux
